### PR TITLE
[DE-423](analytics): Add metrics for artist linkage

### PR DESCRIPTION
**Pull request by lmontier-pass**
# DS PR

## Describe your changes

* Add metric computatiojn at the end of artist linkage
Run on staging : 

## Ticket

https://www.notion.so/passcultureapp/880351701699470cb129e4e0ad6bbcba?v=cfbe4b16778b47218f252bd1b5add4a0&p=8db229c9fd754135a6d130495c400431&pm=s

### Type of change
- [ ] hotfix (non-breaking change which fixes an issue)
- [ ] New model
- [ ] Training
- [ ] New features
- [ ] Bug Fix
- [ ] Code Refacto
- [ ] Performance Improvements
- [x] Test
- [ ] CI
- [ ] Config

      
### Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] My code passes CI/CD tests
- [ ] I updated README.md
- [ ] I have updated the dag
- [ ] If my changes concern incremental table, I have altered their schema to accomodate with field's creation/deletion
- [ ] I have documented the corresponding [notion page](https://www.notion.so/passcultureapp/Team-Data-engineering-Data-science-22ab0eb5ddf34dc2a854d9f0e596e91b)
- [ ] I have made corresponding changes to the [tables documentation](https://www.notion.so/passcultureapp/Documentation-Tables-175a397a8e854ff4a55ae4f3620dbe3b)
- [ ] I have made corresponding changes to the [fields glossary](https://www.notion.so/passcultureapp/854a436a8f1541e1b6ec2a65f8bab600?v=798024ba90404b139e5a17407a3bc604)
- [ ] I will create a review on slack and ensure to specify the duration of the review task: short (<10min), medium (<30min), long (>30min)

### Added tests?
- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] ⏰ no, but I created a ticket

### CODE REVIEW

File: orchestration/dags/jobs/ml/link_artists.py [hunk 1]
```diff
@@ -32,13 +36,21 @@
 PREPROCESSED_GCS_PATH = f"gs://{STORAGE_PATH}/preprocessed_artists_to_match.parquet"
```
Comment (cdelabre) [inline orchestration/dags/jobs/ml/link_artists.py:36]: Update storage path to ML Flow

File: jobs/ml_jobs/artist_linkage/evaluate.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            fp=lambda df: ~df.is_my_artist & df.is_main_cluster,
+            fn=lambda df: df.is_my_artist & ~df.is_main_cluster,
+            tn=lambda df: ~df.is_my_artist & ~df.is_main_cluster,
+        )
+        .groupby("dataset_name")
+        .apply(compute_metrics_per_slice)
+        .reset_index()
+    )
+
+    metrics_per_dataset_df.to_parquet(output_file_path, index=False)
```
Comment (cdelabre) [inline jobs/ml_jobs/artist_linkage/evaluate.py:83]: Prochaine ité - mettre dans ML flow

File: orchestration/dags/dependencies/ml/linkage/sql/artist_linkage_on_test_set.sql [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    artists_to_link.is_synchronised,
+    artists_to_link.artist_type,
+    artists_to_link.offer_number,
+    artists_to_link.total_booking_count,
+    test_set.dataset_name,
+    test_set.is_my_artist,
+    test_set.irrelevant_data
+  FROM
+    `{{ bigquery_tmp_dataset }}`.artists_to_link AS artists_to_link
+  RIGHT JOIN
```
Comment (cdelabre) [inline orchestration/dags/dependencies/ml/linkage/sql/artist_linkage_on_test_set.sql:15]: INVERSE

File: orchestration/dags/jobs/ml/link_artists.py [hunk 4]
```diff
@@ -86,26 +128,16 @@
...
-    install_dependencies = SSHGCEOperator(
-        task_id="install_dependencies",
-        instance_name=GCE_INSTANCE,
-        base_dir=BASE_DIR,
-        command="""pip install -r requirements.txt --user""",
+    collect_test_sets_into_bq = GCSToBigQueryOperator(
+        task_id="import_test_sets_in_bq",
+        bucket=DATA_GCS_BUCKET_NAME,
+        source_objects=[
+            f"{IMPORT_TEST_SET_GCS_BASE_PATH.split(f'{DATA_GCS_BUCKET_NAME}/')[-1]}/*.parquet"
```
Comment (cdelabre) [inline orchestration/dags/jobs/ml/link_artists.py:135]: Create variable instead

File: orchestration/dags/jobs/ml/link_artists.py (hunk 1 above)
Comment (lmontier-pass) [reply orchestration/dags/jobs/ml/link_artists.py:36]: :+1:

File: jobs/ml_jobs/artist_linkage/evaluate.py (hunk 2 above)
Comment (lmontier-pass) [reply jobs/ml_jobs/artist_linkage/evaluate.py:83]: https://www.notion.so/passcultureapp/880351701699470cb129e4e0ad6bbcba?v=cfbe4b16778b47218f252bd1b5add4a0&p=596ca17916e646e095dd724d449f71c2&pm=s

File: orchestration/dags/jobs/ml/link_artists.py [hunk 5]
```diff
@@ -22,28 +23,44 @@
...
+from dependencies.ml.linkage.import_artists import PARAMS as IMPORT_ARTISTS_PARAMS
 
 DEFAULT_REGION = "europe-west1"
 GCE_INSTANCE = f"link-artists-{ENV_SHORT_NAME}"
 BASE_DIR = "data-gcp/jobs/ml_jobs/artist_linkage"
-STORAGE_PATH = f"{DATA_GCS_BUCKET_NAME}/link_artists"
-INPUT_GCS_PATH = f"gs://{STORAGE_PATH}/artists_to_match.parquet"
-PREPROCESSED_GCS_PATH = f"gs://{STORAGE_PATH}/preprocessed_artists_to_match.parquet"
-OUTPUT_GCS_PATH = f"gs://{STORAGE_PATH}/matched_artists.parquet"
+SCHEDULE_CRON = "0 3 * * 1"
```
Comment (cdelabre) [inline orchestration/dags/jobs/ml/link_artists.py:35]: On peut configurer aussi si besoin en fonction des environnements (dev, stg, prod)

**Review by cdelabre — APPROVED** [review]

**Pull request merged**